spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/TestSparkDistributionAndOrderingUtil.java

```diff
@@ -1567,6 +1567,370 @@ public void testRangePositionDeltaUpdatePartitionedTable() {
         table, UPDATE, expectedDistribution, SPEC_ID_PARTITION_FILE_POSITION_ORDERING);
   }
 
+  // ==================================================================================
+  // Distribution and ordering for merge-on-read MERGE operations with position deletes
+  // ==================================================================================
+  //
+  // UNPARTITIONED UNORDERED
+  // -------------------------------------------------------------------------
+  // merge mode is NOT SET -> rely on write distribution and ordering as a basis
+  // merge mode is NONE -> unspecified distribution + LOCALLY ORDER BY _spec_id, _partition, _file, _pos
+  // merge mode is HASH -> unspecified distribution + LOCALLY ORDER BY _spec_id, _partition, _file, _pos
+  // merge mode is RANGE -> unspecified distribution + LOCALLY ORDER BY _spec_id, _partition, _file, _pos
+  //
+  // UNPARTITIONED ORDERED BY id, data
+  // -------------------------------------------------------------------------
+  // merge mode is NOT SET -> rely on write distribution and ordering as a basis
+  // merge mode is NONE -> unspecified distribution +
+  //                       LOCALLY ORDER BY _spec_id, _partition, _file, _pos, id, data
```

Comment:
I think it may be worth an additional comment at the top of this section that says: `IMPORTANT: metadata columns like _spec_id and _partition are null for new rows`.

I think what this means is:
* Deletes and updates will be clustered by original partition and _file, and additionally sorted by _pos
* Inserts will effectively be distributed by id, data and sorted by id, data

This holds roughly true for all of the cases.

Reply:
Good idea, will add a comment.

Reply:
Added a comment.